flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2432,31 +2421,16 @@ struct FieldIndexOpConversion : public FIROpConversion<fir::FieldIndexOp> {
   }
 };
 
+/// Lower fir.len_param_index
 struct LenParamIndexOpConversion
     : public FIROpConversion<fir::LenParamIndexOp> {
   using FIROpConversion::FIROpConversion;
 
   // FIXME: this should be specialized by the runtime target
   mlir::LogicalResult
-  matchAndRewrite(fir::LenParamIndexOp lenp, OperandTy operands,
+  matchAndRewrite(fir::LenParamIndexOp lenp, OpAdaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto ity = lowerTy().indexType();
-    auto onty = lenp.getOnType();
-    // size of portable descriptor
-    const unsigned boxsize = 24; // FIXME
-    unsigned offset = boxsize;
-    // add the size of the rows of triples
-    if (auto arr = onty.dyn_cast<fir::SequenceType>())
-      offset += 3 * arr.getDimension();
-
-    // advance over some addendum fields
-    const unsigned addendumOffset{sizeof(void *) + sizeof(uint64_t)};
-    offset += addendumOffset;
-    // add the offset into the LENs
-    offset += 0; // FIXME
-    auto attr = rewriter.getI64IntegerAttr(offset);
-    rewriter.replaceOpWithNewOp<mlir::LLVM::ConstantOp>(lenp, ity, attr);
-    return success();
+    TODO(lenp.getLoc(), "fir.len_param_index codegen");
```

Comment:
Diverge here as well with the caveat that a hard TODO should still be added here. The LLVM IR is incorrect, but there was only a comment, so it would fail silently and produce a bad program.

Reply:
Done.